lib/github_changelog_generator/generator/entry.rb

```diff
@@ -0,0 +1,232 @@
+require "github_changelog_generator/generator/section"
+
+module GitHubChangelogGenerator
+  class Entry
+    attr_reader :content
+
+    def initialize(options = Options.new({}))
+      @content = ""
+      @options = Options.new(options)
+    end
+
+    # Generates log entry with header and body
+    #
+    # @param [Array] pull_requests List or PR's in new section
+    # @param [Array] issues List of issues in new section
+    # @param [String] newer_tag_name Name of the newer tag. Could be nil for `Unreleased` section.
+    # @param [String] newer_tag_link Name of the newer tag. Could be "HEAD" for `Unreleased` section.
+    # @param [Time] newer_tag_time Time of the newer tag
+    # @param [Hash, nil] older_tag Older tag, used for the links. Could be nil for last tag.
+    # @return [String] Ready and parsed section
+    def create_entry_for_tag(pull_requests, issues, newer_tag_name, newer_tag_link, newer_tag_time, older_tag_name) # rubocop:disable Metrics/ParameterLists
+      github_site = @options[:github_site] || "https://github.com"
+      project_url = "#{github_site}/#{@options[:user]}/#{@options[:project]}"
+
+      set_sections_and_maps
+
+      @content = generate_header(newer_tag_name, newer_tag_link, newer_tag_time, older_tag_name, project_url)
+
+      @content += generate_body(pull_requests, issues)
+
+      @content
+    end
+
+    private
+
+    # Creates section objects and the label and section maps needed for
+    # sorting
+    def set_sections_and_maps
+      @sections = if @options.configure_sections?
+                    parse_sections(@options[:configure_sections])
+                  elsif @options.add_sections?
+                    default_sections.concat parse_sections(@options[:add_sections])
+                  else
+                    default_sections
+                  end
+
+      @lmap = label_map
+      @smap = section_map
+    end
+
+    # Turns a string from the commandline into an array of Section objects
+    #
+    # @param [String, Hash] either string or hash describing sections
+    # @return [Array] array of Section objects
+    def parse_sections(sections_desc)
+      require "json"
+
+      sections_desc = sections_desc.to_json if sections_desc.class == Hash
+
+      begin
+        sections_json = JSON.parse(sections_desc)
+      rescue JSON::ParserError => e
+        raise "There was a problem parsing your JSON string for secions: #{e}"
+      end
+
+      sections_arr = []
+
+      sections_json.each do |name, v|
+        sections_arr << Section.new(name: name.to_s, prefix: v["prefix"], labels: v["labels"], options: @options)
+      end
+
+      sections_arr
```

Comment:
We can leave off naming a local variable here, and only return the result of `#map`

```ruby
sections_json.map do |name, v|
  Section.new(name: name.to_s, prefix: v["prefix"],
              labels: v["labels"], options: @options)
end 
```